sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -77,7 +82,21 @@
 
         boolean eagerlyReadResponse = (boolean) context.getData("azure-eagerly-read-response").orElse(false);
 
-        return nettyClient
+        reactor.netty.http.client.HttpClient client = this.nettyClient;
+        Optional<Object> requestResponseTimeout = context.getData(AZURE_RESPONSE_TIMEOUT);
+        if (requestResponseTimeout.isPresent()) {
+            client = nettyClient.doAfterRequest((req, connection) -> {
+                // remove write timeout handler and client-level response timeout handler before adding request-level
+                // response timeout handler
+                long responseTimeoutMillis = ((Duration) requestResponseTimeout.get()).toMillis();
+                connection.removeHandler(WriteTimeoutHandler.HANDLER_NAME)
+                    .removeHandler(ResponseTimeoutHandler.HANDLER_NAME)
+                    .addHandlerLast(ResponseTimeoutHandler.HANDLER_NAME,
```

Comment:
Maybe it would be better to make the `doAfterRequest` handler a method which accepts a `ResponseTimeoutHandler` and if `AZURE_RESPONSE_TIMEOUT` isn't configured it uses the default one in the builder. This would be better as we aren't creating a handler that will be thrown away anymore.